datetime/localtimedb.c

```diff
@@ -2207,13 +2208,9 @@ static struct db_state *find_tz(const char *name)
 
 static void add_tz(const char *name, struct db_state *ptr)
 {
-    char key[NAME_KEY_MAX];
-    tz_hash_entry_type *hash_entry_ptr;
-
-    bzero(key, NAME_KEY_MAX);
-    strcpy(key, name);
-
-    hash_entry_ptr = malloc(sizeof(tz_hash_entry_type));
+    char key[NAME_KEY_MAX] = {0};
+    strncpy0(key, name, sizeof(key));
```

Comment:
I don't think we need `key` here. We could copy `name` directly into `hash_entry_ptr->key`.

Reply:
Good point.